src/github.com/matrix-org/dendrite/syncapi/sync/requestpool.go

```diff
@@ -77,9 +80,14 @@ func (rp *RequestPool) OnIncomingSyncRequest(req *http.Request, device *authtype
 		if err != nil {
 			res = httputil.LogThenError(req, err)
 		} else {
-			res = util.JSONResponse{
-				Code: 200,
-				JSON: syncData,
+			syncData, err = rp.appendAccountData(syncData, device.UserID)
```

Comment:
Are we including all the account data to every sync response? It might be worth adding a TODO marker to say that we should include the data that has changed for incremental sync responses.